tests/unit/misc/test_overlap.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Tests for qutebrowser.misc.overlap."""
+
+import pytest
+
+from PyQt5.QtWidgets import QWidget
+
+from qutebrowser.misc import overlap
+
+
+def test_overlap1(qtbot):
+    """Overlap should be found between two widgets."""
+    parent = QWidget()
+    parent.resize(100, 100)
+    qtbot.add_widget(parent)
+
+    widget1 = QWidget(parent=parent)
+    widget1.resize(20, 10)
+    widget1.move(0, 0)
+    widget2 = QWidget(parent=parent)
+    widget2.resize(20, 10)
+    widget2.move(10, 0)
+
+    overlap_lookup = overlap.OverlapLookup([widget1, widget2])
+    assert len(overlap_lookup.overlappings(widget1)) == 1
+
+
+@pytest.fixture()
+def three_widgets(qtbot):
+    parent = QWidget()
+    parent.resize(100, 100)
+    qtbot.add_widget(parent)
+    parent.show()
+
+    widget1 = QWidget(parent=parent)
+    widget1.resize(20, 10)
+    widget1.move(0, 0)
+    widget1.show()
+    widget2 = QWidget(parent=parent)
+    widget2.resize(20, 10)
+    widget2.move(10, 0)
+    widget2.show()
+    widget3 = QWidget(parent=parent)
+    widget3.resize(20, 10)
+    widget3.move(25, 5)
+    widget3.show()
+
+    # have to return parent or objects will be deleted
```

Comment:
If you use the `qapp` fixture and set `parent=qapp` for the parent, it should not be deleted even if you only return the children here.

Reply:
Please modify as you see fit.